rafx-shader-processor/src/parse_source.rs

```diff
@@ -160,9 +202,26 @@ pub(crate) fn try_consume_array_index(
 
         // If this fails, then we may have a string like "123xyz"
         let number: usize = characters_to_string(&code[begin..end]).parse().ok()?;
-
         *position = end;
         Some(number)
+    } else if begin < code.len() {
+        let end = next_non_identifer(code, begin);
+        let identifier = characters_to_string(&code[begin..end]);
```

Comment:
I think if you have an array float my_array[ABC * XYZ], it will return the value of ABC and ignore the * XYZ part.

Doing this with math operators supported would probably require us to do the arithmetic with correct order of operation (or require explicit parenthesis). As a first pass, maybe we can require that constants are literal, numeric constants.